meta-lmp-base/classes/fip-utils.bbclass

```diff
@@ -0,0 +1,391 @@
+                if [ -n "${TF_A_SOC_NAME}" ]; then
+                    unset k
+                    for soc in ${TF_A_SOC_NAME}; do
+                        k=$(expr $k + 1)
+                        if [ "$(echo ${dt} | grep -c ${soc})" -eq 1 ]; then
+                            sign_key=$(echo ${FIP_SIGN_KEY_PATH_SOC_LIST} | cut -d',' -f${k})
+                        fi
+                    done
+                else
+                    sign_key="${FIP_SIGN_KEY_PATH}"
+                fi
+                if [ -z "${sign_key}" ]; then
+                    bbfatal "Please make sure to configure \"FIP_SIGN_KEY\" var to signing key file."
+                fi
+                FIP_CERTCONF="\
+                    --tb-fw-cert ${WORKDIR}/tb_fw.crt \
+                    --trusted-key-cert ${WORKDIR}/trusted_key.crt \
+                    --nt-fw-cert ${WORKDIR}/nt_fw_content.crt \
+                    --nt-fw-key-cert ${WORKDIR}/nt_fw_key.crt \
+                    --tos-fw-cert ${WORKDIR}/tos_fw_content.crt \
+                    --tos-fw-key-cert ${WORKDIR}/tos_fw_key.crt \
+                    "
+                # Need fake bl2 binary to generate certificates
+                touch ${WORKDIR}/bl2-fake.bin
+                # Generate certificates
+                ${CERTTOOL} -n --tfw-nvctr 0 --ntfw-nvctr 0 --key-alg ecdsa --hash-alg sha256 \
+                        --rot-key ${sign_key} \
+                        --rot-key-pwd ${FIP_SIGN_KEY_PASS} \
+                        ${FIP_FWCONFIG} \
+                        ${FIP_HWCONFIG} \
+                        ${FIP_NTFW} \
+                        ${FIP_EXTRACONF} \
+                        ${FIP_CERTCONF} \
+                        --tb-fw ${WORKDIR}/bl2-fake.bin
+                # Remove fake bl2 binary
+                rm -f ${WORKDIR}/bl2-fake.bin
+            else
+                FIP_CERTCONF=""
+            fi
+            # Generate FIP binary
+            bbnote "${FIPTOOL} create \
+                            ${FIP_FWCONFIG} \
+                            ${FIP_HWCONFIG} \
+                            ${FIP_NTFW} \
+                            ${FIP_BL31CONF} \
+                            ${FIP_EXTRACONF} \
+                            ${FIP_CERTCONF} \
+                            ${FIP_DEPLOYDIR_FIP}/${FIP_BASENAME}-${dt}-${config}${FIP_SIGN_SUFFIX}.${FIP_SUFFIX}"
+            ${FIPTOOL} create \
+                            ${FIP_FWCONFIG} \
+                            ${FIP_HWCONFIG} \
+                            ${FIP_NTFW} \
+                            ${FIP_BL31CONF} \
+                            ${FIP_EXTRACONF} \
+                            ${FIP_CERTCONF} \
+                            ${FIP_DEPLOYDIR_FIP}/${FIP_BASENAME}-${dt}-${config}${FIP_SIGN_SUFFIX}.${FIP_SUFFIX}
+        done
+    done
+}
+
+# Stub do_compile for nativesdk use case as we only expect to provide FIPTOOL_WRAPPER
+do_compile:class-nativesdk() {
+    return
+}
+
+do_install:class-nativesdk() {
```

Comment:
We can probably drop class-nativesdk from here as well, not something we currently use or support, so better to add later if needed.